sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -179,16 +185,16 @@ class UnivocityParser(
       dataType: DataType,
       nullable: Boolean = true): ValueConverter = dataType match {
     case _: ByteType => (d: String) =>
-      nullSafeDatum(d, name, nullable, options)(_.toByte)
+      nullSafeDatum(d, name, nullable, options)(retryWithTrim[Byte](_.toByte))
```

Comment:
the performance will be terrible if most data has whitespace, as now we pay the per-row cost of error handling.

Reply:
Currently, I haven't come up with a better solution.

Reply:
Maybe this isn't an issue, as people can turn on `ignoreLeadingWhiteSpace`.

Reply:
Okay, when the end-user sees a return value of `NULL`, we don't seem to have `a better way` to notice them (if an exception occurs, we may also notice them in the error message, but in our scenario, it won't). Alright, let's leave it for now. 
Thanks!

Reply:
I even suspect it's a problem with JDK itself, haha
  
A.Int: java.lang.Integer.parseInt(...)
  ```
  scala> val x = java.lang.Integer.parseInt(" 1")
  java.lang.NumberFormatException: For input string: " 1"
    at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
    at java.base/java.lang.Integer.parseInt(Integer.java:654)
    at java.base/java.lang.Integer.parseInt(Integer.java:786)
    ... 41 elided
  ```

  B.Double: java.lang.Double.parseDouble(...)
  ```scala
  scala> val x = java.lang.Double.parseDouble(" 1")
  val x: Double = 1.0
  ```

  ```java
  public static double parseDouble(String s) throws NumberFormatException {
    return FloatingDecimal.parseDouble(s);
  }
  ```

  <img width="800" alt="image" src="https://github.com/user-attachments/assets/406ca81f-2d11-4a1a-9f42-bd8e5ffab03f">